### PR TITLE
Tidy up intents

### PR DESCRIPTION
**Pull request by deanrobertcook**
Tidies up how intents are handled by the main activity (a bit). Should also fix the bug where content can be shared into the wrong conversation
#### APK
[Download build #9478](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9478/artifact/build/artifact/wire-dev-PR1130-9478.apk)
[Download build #9479](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9479/artifact/build/artifact/wire-dev-PR1130-9479.apk)
[Download build #9480](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9480/artifact/build/artifact/wire-dev-PR1130-9480.apk)
[Download build #9481](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9481/artifact/build/artifact/wire-dev-PR1130-9481.apk)
[Download build #9482](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9482/artifact/build/artifact/wire-dev-PR1130-9482.apk)
[Download build #9483](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9483/artifact/build/artifact/wire-dev-PR1130-9483.apk)
[Download build #9484](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9484/artifact/build/artifact/wire-dev-PR1130-9484.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/MainActivity.scala [hunk 1]
```diff
@@ -338,68 +320,68 @@ class MainActivity extends BaseActivity
...
-
-            getControllerFactory.getSharingController.setSharedText(sharedText)
-            getControllerFactory.getSharingController.setSharedUris(sharedFileUris)
-            getControllerFactory.getSharingController.setSharingConversationId(convId)
+  def handleIntent(intent: Intent) = {
+    verbose(s"handleIntent: ${intent.log}")
 
-            Option(conv).foreach(_.setEphemeralExpiration(expiration))
+    def switchConversation(convId: ConvId, call: Boolean = false, exp: EphemeralExpiration = EphemeralExpiration.NONE) =
+      CancellableFuture.delay(125.millis).map { _ =>
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/MainActivity.scala]: Why exactly 125 millis and why is it hardcoded?

**Comment by someoneEsle** [comment]
Bug 1
After sharing to contact A, the conversation with A should open.

Bug 2
Opening Wire from quick reply does not open the conversation. However, it does open the conversations list from the right account.

Bug 3
Tapping notifications only opens the conversations list without taking the user to the conversation.

**Comment by deanrobertcook** [comment]
@someoneEsle all should be fixed now

**Comment by someoneEsle** [comment]
Bug 1 is fixed. Bug 2 and 3 are still there.

**Comment by someoneEsle** [comment]
Bug 4
Sometimes when trying to call someone by tapping the Call button in the notification, you call the second person in your conversations list.

**Comment by someoneEsle** [comment]
Bug 4 is not related to this PR, so I can remove the QA label here.

**Pull request merged**